web/modules/custom/joinup_discussion/src/EventSubscriber/SubscribedDiscussionCommentSubscriber.php

```diff
@@ -0,0 +1,195 @@
+   */
+  protected $group;
+
+  /**
+   * The list of recipients as user accounts.
+   *
+   * @var \Drupal\user\UserInterface[]
+   */
+  protected $recipients;
+
+  /**
+   * Constructs a new event subscriber object.
+   *
+   * @param \Drupal\joinup_subscription\JoinupSubscriptionInterface $subscribe_service
+   *   The Joinup subscribe service.
+   * @param \Drupal\joinup_notification\JoinupMessageDeliveryInterface $message_delivery
+   *   The Joinup message delivery service.
+   */
+  public function __construct(JoinupSubscriptionInterface $subscribe_service, JoinupMessageDeliveryInterface $message_delivery) {
+    $this->subscribeService = $subscribe_service;
+    $this->messageDelivery = $message_delivery;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    return [NotificationEvents::COMMENT_CRUD => 'commentCrudProxy'];
+  }
+
+  /**
+   * Acts as a proxy by passing the control to a dedicated method.
+   *
+   * Normally we should have split the events into more specific events, like
+   * 'comment create', 'comment update', 'comment delete'. Just 'comment crud'
+   * is too generic and is not actually an event. In order to keep the
+   * compatibility with Joinup Notification module, we use this proxy method to
+   * delegate the reaction to a specific protected method, based on the real
+   * operation being performed.
+   *
+   * @param \Drupal\joinup_notification\Event\NotificationEvent $event
+   *   The notification event object.
+   */
+  public function commentCrudProxy(NotificationEvent $event): void {
+    $this->comment = $event->getEntity();
+
+    // Discussion comments are 'reply' comment types.
+    if ($this->comment->bundle() !== 'reply') {
+      return;
+    }
+
+    // Orphan comment?
+    if (!$this->discussion = $this->comment->getCommentedEntity()) {
+      return;
+    }
+
+    // Don't handle inconsistent discussions, without a parent group.
+    if (!$this->group = $this->discussion->get('og_audience')->entity) {
+      return;
+    }
+
+    // No recipients, no reaction.
+    if (!$this->recipients = $this->buildRecipientList()) {
+      return;
+    }
+
+    // Call the appropriate method.
+    switch ($event->getOperation()) {
+      case 'create':
+        $this->onCommentCreate();
+        break;
+
+      case 'update':
+        $this->onCommentUpdate();
+        break;
+    }
+  }
+
+  /**
+   * Reacts when a new comment is added to a discussion.
+   */
+  protected function onCommentCreate(): void {
+    // Do not send the notification if the comment is not yet published. This
+    // normally happens when an anonymous user is posting a comment because such
+    // comments are subject of approval.
+    if ($this->comment->isPublished()) {
+      // Deliver the notification message.
+      $this->messageDelivery
+        ->createMessage('discussion_comment_new')
+        ->setArguments($this->getArguments())
+        ->setRecipients($this->recipients)
```

Comment:
If we also call a method here like `::getRecipients()` instead of relying on the fact that the list of recipient was already built in a previous step this class would become more resilient to bugs if it is expanded or refactored in the future.